mlflow/utils/autologging_utils/safety.py

```diff
@@ -542,7 +560,32 @@ def call_original(*og_args, **og_kwargs):
                 else:
                     return original(*args, **kwargs)
 
-    new_patch = _wrap_patch(destination, function_name, safe_patch_function)
+    if is_property_method:
+
+        def get_bound_safe_patch_fn(self):
+            # This `raw_original_obj.fget` call is for availability check, if it raise error
+            # then `hasattr(obj, {func_name})` will return False
+            # so it mimic the original property behavior.
+            raw_original_obj.fget(self)
+
+            def bound_safe_patch_fn(*args, **kwargs):
+                return safe_patch_function(self, *args, **kwargs)
+
+            # Make bound method `instance.target_method` keep the same doc and signature
+            bound_safe_patch_fn = update_wrapper_extended(
+                bound_safe_patch_fn, raw_original_obj.fget
+            )
+            return bound_safe_patch_fn
```

Comment:
Can we add a comment explaining the distinction between bound and unbound methods and also explain why we need to create a bound version of the safe patch function in order to handle property methods?